internal/controller/nonadminbackup_controller.go

```diff
@@ -70,167 +62,46 @@ const (
-		// Regardless if the status was updated or not, we should not
-		// requeue here as it was only status update.
-		if errBackupUpdate != nil {
-			return ctrl.Result{}, errBackupUpdate
-		} else if updatedNab {
-			logger.V(1).Info("NonAdminBackup CR - Rqueue after Status Update")
-			return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, nil
-		}
-		return ctrl.Result{}, nil
 	}
 
-	// Ensure labels are set for the Backup object
-	existingLabels := veleroBackup.Labels
-	naManagedLabels := function.AddNonAdminLabels(existingLabels)
-	veleroBackup.Labels = naManagedLabels
-
-	// Ensure annotations are set for the Backup object
-	existingAnnotations := veleroBackup.Annotations
-	ownerUUID := string(nab.ObjectMeta.UID)
-	nabManagedAnnotations := function.AddNonAdminBackupAnnotations(nab.Namespace, nab.Name, ownerUUID, existingAnnotations)
-	veleroBackup.Annotations = nabManagedAnnotations
-
-	_, err = controllerutil.CreateOrPatch(ctx, r.Client, &veleroBackup, nil)
-	if err != nil {
-		logger.Error(err, "Failed to create backup", nameField, veleroBackupName)
-		return ctrl.Result{}, err
-	}
-	logger.Info("VeleroBackup successfully created", nameField, veleroBackupName)
-
-	// Phase: Created
-	// BackupAccepted: True # do not update
-	// BackupQueued: True
-	_, errUpdate = function.UpdateNonAdminPhase(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminBackupPhaseCreated)
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: Created", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
-	_, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionTrue, "Validated", "Valid Backup config")
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set BackupAccepted Condition: True", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
-	_, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionQueued, metav1.ConditionTrue, "BackupScheduled", "Created Velero Backup object")
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set BackupQueued Condition: True", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
+	// Run Reconcile Batch as we have the NonAdminBackup object
+	reconcileExit, reconcileRequeue, reconcileErr := ReconcileBatch(
+		ReconcileFunc(func(...any) (bool, bool, error) {
+			return r.InitNonAdminBackup(ctx, rLog, &nab) // Phase: New
+		}),
+		ReconcileFunc(func(...any) (bool, bool, error) {
+			return r.ValidateVeleroBackupSpec(ctx, rLog, &nab) // Phase: New
+		}),
+		ReconcileFunc(func(...any) (bool, bool, error) {
+			return r.CreateVeleroBackupSpec(ctx, rLog, &nab) // Phase: New
+		}),
+	)
+
+	// Return vars from the ReconcileBatch
+	if reconcileExit {
```

Comment:
would also simplify here
```go
	if reconcileErr != nil  || reconcileExit {
		return ctrl.Result{}, reconcileErr
	}
        if reconcileRequeue {
		return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, reconcileErr
	}
```

Reply:
How about even simpler and entire thing replace with. See we don't really care here about reconcileExit anymore, because this is the same as the last line of the reconcile loop ?:

```
	// Run Reconcile Batch as we have the NonAdminBackup object
	_, reconcileRequeue, reconcileErr := ReconcileBatch(
		ReconcileFunc(func(...any) (bool, bool, error) {
			return r.InitNonAdminBackup(ctx, rLog, &nab) // Phase: New
		}),
		ReconcileFunc(func(...any) (bool, bool, error) {
			return r.ValidateVeleroBackupSpec(ctx, rLog, &nab) // Phase: New
		}),
		ReconcileFunc(func(...any) (bool, bool, error) {
			return r.CreateVeleroBackupSpec(ctx, rLog, &nab) // Phase: New
		}),
	)

	// Return vars from the ReconcileBatch
	if reconcileRequeue {
		return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, reconcileErr
	}

	logger.V(1).Info(">>> Reconcile NonAdminBackup - loop end")
	return ctrl.Result{}, reconcileErr
}
```

Reply:
Adjusted last commit with the above change. Note we do have now `_` as the return from ReconcileBatch, we may want to either remove first bool or keep it depending on how other controllers will utilize that function.

Reply:
See the oadp has similar `_` implemented: https://github.com/openshift/oadp-operator/blob/master/controllers/dpa_controller.go#L93

Reply:
and it should be removed as well https://github.com/search?q=repo%3Aopenshift%2Foadp-operator+ReconcileBatch+language%3AGo&type=code&l=Go 

exit does the same job as a return (exit this function early, with or without error). It should be removed

Reply:
Ok I have very much reworked that part and this comment is not relevant anymore with the new commit which is not using ReconcileBatch and we need args there.